numpy/testing/_private/utils.py

```diff
@@ -920,25 +935,26 @@ def assert_array_almost_equal(x, y, decimal=6, err_msg='', verbose=True):
     the first assert does not raise an exception
 
     >>> np.testing.assert_array_almost_equal([1.0,2.333,np.nan],
-                                             [1.0,2.333,np.nan])
+    ...                                      [1.0,2.333,np.nan])
 
     >>> np.testing.assert_array_almost_equal([1.0,2.33333,np.nan],
     ...                                      [1.0,2.33339,np.nan], decimal=5)
     ...
-    <type 'exceptions.AssertionError'>:
     AssertionError:
-    Arrays are not almost equal
-    (mismatch 50.0%)
+    Arrays are not almost equal to 5 decimals
+    Mismatch: 33.3%
+    Max absolute difference: 6.e-05
+    Max relative difference: 2.57136612e-05
      x: array([ 1.     ,  2.33333,      NaN])
      y: array([ 1.     ,  2.33339,      NaN])
 
     >>> np.testing.assert_array_almost_equal([1.0,2.33333,np.nan],
     ...                                      [1.0,2.33333, 5], decimal=5)
-    <type 'exceptions.ValueError'>:
-    ValueError:
-    Arrays are not almost equal
-     x: array([ 1.     ,  2.33333,      NaN])
-     y: array([ 1.     ,  2.33333,  5.     ])
+    AssertionError:
+    Arrays are not almost equal to 5 decimals
+    x and y nan location mismatch:
+     x: array([1.     , 2.33333,     nan])
+     y: array([1.     , 2.33339, 5.     ])
```

Comment:
refguide is opt-in so if @shoyer thinks we should be running the checker on these docstrings then:

```diff
diff --git a/tools/refguide_check.py b/tools/refguide_check.py
index 3d885e3..fec715f 100644
--- a/tools/refguide_check.py
+++ b/tools/refguide_check.py
@@ -76,6 +76,7 @@ PUBLIC_SUBMODULES = [
     'polynomial',
     'matrixlib',
     'random',
+    'testing',
 ]
```

Testing locally (`python runtests.py -g --refguide-check >& refguide_results.txt`), this causes various failures, some for the docstrings touched here, but also for other `testing/_private/utils.py` docstrings with things like `suppress_warnings is not defined`.

I wonder if there's an extra layer of complexity given that we are testing the testing utilities' docstrings. Normally refguide failures will produce an `expected` and `got` set of values, but many of the problems here seem to complain about the desired assertion actually happening (something not being caught when it should be maybe).

My initial bet is that this is the design issue observed with refguide here:
```python
    def _do_check(self, want, got):
        # This should be done exactly as written to correctly handle all of
        # numpy-comparable objects, strings, and heterogeneous tuples
        try:
            if want == got:
                return True
        except Exception:
            pass
        return np.allclose(want, got, atol=self.atol, rtol=self.rtol)
```

In particular, I suppose we'd want a way to check if `allclose` and / or similar are the actual function under test in the docstring and then perform a meta-assessment from there. I find this slightly mind-bending, but maybe obvious to other devs.

That said, is this something for a separate issue / wish-list? Or maybe simpler than I'm thinking here.

Reply:
I was able to successfully enable refguide checking on `np.testing` with a few docstring fixes. It appears the trick is add to include lines for the trraceback at the start of each error message:
```
Traceback (most recent call last):
    ...
```
Otherwise doctests don't recognize the exception.